src/components/ActionMenu/components/Actions/Actions.tsx

```diff
@@ -35,32 +56,92 @@ export function Actions({actions = [], groups = []}: Props) {
     () => setActiveMenuGroup(undefined),
     [],
   );
-  const menuActions = [...actions, ...groups];
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleResize = useCallback(
+    debounce(
+      () => {
+        if (!newDesignLanguage || !actionsLayoutRef.current) return;
+        setAvailableWidth(actionsLayoutRef.current.offsetWidth);
+      },
+      20,
+      {leading: false, trailing: true, maxWait: 40},
+    ),
+    [actionsLayoutRef],
+  );
+
+  useEffect(() => {
+    if (!actionsLayoutRef.current) return;
+    setAvailableWidth(actionsLayoutRef.current.offsetWidth);
+  }, [actionsLayoutRef]);
+
+  useEffect(() => {
+    if (!menuGroupRef.current) return;
+    menuGroupWidthRef.current = menuGroupRef.current?.offsetWidth;
+  }, [menuGroupRef]);
+
+  useEffect(() => {
+    if (!newDesignLanguage || actionWidths.length === 0 || availableWidth === 0)
+      return;
+
+    let currentAvailableWidth = availableWidth;
+    setShowableActions([]);
+    setHiddenActions([]);
+
+    actions.forEach((action, index) => {
+      if (
+        actionWidths[index] + menuGroupWidthRef.current + ACTION_SPACING <=
+        currentAvailableWidth
+      ) {
+        currentAvailableWidth -= actionWidths[index];
+        setShowableActions((showableActions) => [
+          ...(showableActions || []),
+          action,
+        ]);
+      } else {
+        currentAvailableWidth = 0;
```

Comment:
I'm doing this because when there's a single item that can't fit we roll up the remaining items. If we don't set this to 0 then we might see actions jumping around in different positions on different viewport sizes. 

i.e. if there is an action in the array that is smaller than a previous action that couldn't fit, it may show up in its place